### PR TITLE
Spark/transformation type masking

### PR DESCRIPTION
**Pull request by tnazarew**
### Problem

Spark integration doesn't have a mechanism to extract information about whether or not function is masking.

### Solution
Implement mechanism to extract imformation about masking expressions.

#### One-line summary:

Spark integration can't extract information about `masking` property value of the field, so implement it

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [x] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [x] Your comment includes a one-liner for the changelog about the specific purpose of the change (_not required for changes to tests, docs, or CI config_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2024 contributors to the OpenLineage project

### CODE REVIEW

File: integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/column/ExpressionDependencyCollector.java [hunk 1]
```diff
@@ -51,6 +59,19 @@
  */
 @Slf4j
 public class ExpressionDependencyCollector {
+  private static Boolean isMasking(Expression expression) {
+
+    return expression instanceof Crc32
```
Comment (pawel-big-lebowski) [inline integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/column/ExpressionDependencyCollector.java:64]: Wouldn't it look better if there was a constant list `List<Class>` storing masking functions? 
What do you think?

File: integration/spark/shared/src/main/java/io/openlineage/spark/agent/lifecycle/plan/column/TransformationInfo.java [hunk 2]
```diff
@@ -40,15 +40,27 @@ public static TransformationInfo identity() {
...
-    return new TransformationInfo(Types.DIRECT, Subtypes.TRANSFORMATION, "", false);
+    return transformation(false);
+  }
+
+  public static TransformationInfo transformation(Boolean isMasking) {
+    return new TransformationInfo(Types.DIRECT, Subtypes.TRANSFORMATION, "", isMasking);
   }
 
   public static TransformationInfo aggregation() {
-    return new TransformationInfo(Types.DIRECT, Subtypes.AGGREGATION, "", false);
```
Comment (pawel-big-lebowski) [inline integration/spark/shared/src/main/java/io/openlineage/spark/agent/lifecycle/plan/column/TransformationInfo.java:47]: Could you provide some docs for those method explaining/giving an example on how an aggregation transformation type can be masking or not?

**Review by pawel-big-lebowski — APPROVED** [review]
Great work @tnazarew Two comments added.

File: integration/spark/shared/src/main/java/io/openlineage/spark/agent/lifecycle/plan/column/TransformationInfo.java (hunk 2 above)
Comment (tnazarew) [reply integration/spark/shared/src/main/java/io/openlineage/spark/agent/lifecycle/plan/column/TransformationInfo.java:47]: well the only one masking, aggregation transformation I could think of is `count` as it completely disregards the value in the column no matter what

Comment (tnazarew) [reply integration/spark/shared/src/main/java/io/openlineage/spark/agent/lifecycle/plan/column/TransformationInfo.java:47]: added some javadoc

**Review by pawel-big-lebowski — APPROVED** [review]

**Pull request merged**